vendor/src/github.com/docker/libnetwork/drivers/bridge/port_mapping.go

```diff
@@ -14,7 +14,7 @@ var (
 	defaultBindingIP = net.IPv4(0, 0, 0, 0)
 )
 
-func (n *bridgeNetwork) allocatePorts(ep *bridgeEndpoint, reqDefBindIP net.IP, ulPxyEnabled bool) ([]types.PortBinding, error) {
+func (n *bridgeNetwork) allocatePorts(ep *bridgeEndpoint, reqDefBindIP net.IP, ulPxyEnabled bool, pxyBin string) ([]types.PortBinding, error) {
```

Comment:
Instead of passing the custom proxy binary in all the functions up to `portmapper Map()`, could you instead add a new constructor in portmapper package which takes the custom proxy binary path ?
Or maybe change the existing `portmapper.New()` to `New(pxyBin string)`.